guides/v2.2/coding-standards/technical-guidelines.md

```diff
@@ -532,7 +532,75 @@ class View extends Template
 
 ### 6.4. Service Contracts (Application) layer
 
-We are reviewing this section and will publish it soon.
+6.4.1. Location
+
+6.4.1.1. Service contract interfaces should be placed in separate API modules. All other modules will depend on the API module, while implementations could be easily swapped via `di.xml`. For example, if module is named `MyModule`, its APIs should be declared in the module named `MyModuleApi`.
+
+6.4.1.2. Service interfaces which should be exposed as web APIs must be placed under `MyModuleApi/Api/Service` directory. Service data interfaces must be placed under `MyModuleApi/Api/Data`. 
```

Comment:
Does API folder still makes sense for api modules? Maybe we should have something like below?
MyModuleApi\Data
MyModuleApi\Service
MyModuleApi\Model

Reply:
I don't see strong arguments for either of the approaches. Considering existence of `Api` directory in all modules currently, I'm inclined towards this option. It will make transition to the new style easier.

BTW, API modules might have more directories in the root, please take a look at https://github.com/magento-engcom/msi/tree/2.3-develop/app/code/Magento/InventoryApi as an example